clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1069,10 +1069,22 @@ static bool buildArgTys(ASTContext &Context, CXXRecordDecl *KernelObj,
           continue;
         }
       }
-      if (!ArgTy.isTriviallyCopyableType(Context)) {
+
+      CXXRecordDecl *RD =
+          cast<CXXRecordDecl>(ArgTy->getAs<RecordType>()->getDecl());
```

Comment:
Is `cast` usage safe here?
We check `ArgTy->isStructureOrClassType()` below, so is it possible that ArgTy will actually represent a `RecordDecl` but not a `CXXRecordDecl`?

Reply:
RecordDecl only appears in C, all records in C++ mode are CXXRecordDecls.